src/ModelContextProtocol.AspNetCore/IdleTrackingBackgroundService.cs

```diff
@@ -4,16 +4,18 @@
 
 namespace ModelContextProtocol.AspNetCore;
 
-internal sealed partial class IdleTrackingBackgroundService(
-    StatefulSessionManager sessions,
-    IOptions<HttpServerTransportOptions> options,
-    IHostApplicationLifetime appLifetime,
-    ILogger<IdleTrackingBackgroundService> logger) : BackgroundService
+internal sealed partial class IdleTrackingBackgroundService : BackgroundService
 {
-    // Workaround for https://github.com/dotnet/runtime/issues/91121. This is fixed in .NET 9 and later.
-    private readonly ILogger _logger = logger;
+    private readonly StatefulSessionManager _sessions;
+    private readonly IOptions<HttpServerTransportOptions> _options;
+    private readonly IHostApplicationLifetime _appLifetime;
+    private readonly ILogger _logger;
 
-    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    public IdleTrackingBackgroundService(
```

Comment:
I take it you moved away from using a primary ctor because you didn't want to capture the parameters in `ExecuteAsync`? Or was it a different reason?

Reply:
Nevermind, I misunderstood what the change was doing.